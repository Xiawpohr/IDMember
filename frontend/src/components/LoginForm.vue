<template>
  <v-card width="450">
    <v-container>
      <h2 class="display-2 text-xs-center pb-4">Log In</h2>
      <v-form
        v-model="valid"
        @submit.prevent="$emit('submitted', auth)"
      >
        <v-text-field
          v-model="email"
          :rules="[rules.email]"
          label="E-mail"
          color="teal"
          box
          required
        />
        <v-text-field
          v-model="password"
          :rules="[rules.password]"
          label="Password"
          color="teal"
          type="password"
          box
          required
        />
        <div class="text-xs-center">
          <v-btn
            type="submit"
            color="teal"
            large
          >
            Log In
          </v-btn>
        </div>
      </v-form>
    </v-container>
  </v-card>
</template>

<script>
export default {
  data() {
    return {
      email: '',
      password: '',
      valid: false,
      rules: {
        email: v => {
          const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
          return pattern.test(v) || 'Please enter a valid email'
        },
        password: v => {
          const pattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*(_|[^\w])).+$/
          return pattern.test(v) || 'Password must contain an upper case letter, a numeric character, and a special character'
        }
      }
    }
  },
  computed: {
    auth() {
      return {
        email: this.email,
        password: this.password
      }
    }
  }
}
</script>
