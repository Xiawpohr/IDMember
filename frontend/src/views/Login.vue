<template>
  <v-layout justify-center align-center style="height: 100vh">
    <v-progress-circular
      v-if="isLoading"
      :size="150"
      :width="20"
      color="teal"
      indeterminate
    />
    <div v-else>
      <LoginForm @submitted="login" />
      <div class="mt-3 text-xs-center">
        <v-btn
          to='/signup'
          flat
          color="teal"
        >
          Go to Sign Up
        </v-btn>
      </div>
    </div>
  </v-layout>
</template>

<script>
import LoginForm from '@/components/LoginForm.vue'

export default {
  components: {
    LoginForm
  },
  computed: {
    isLoading() {
      return this.$store.state.auth.isLoading
    },
    userId() {
      return this.$store.state.auth.id
    }
  },
  watch: {
    userId(val) {
      if (val !== null) {
        this.$router.push('/')
      }
    }
  },
  methods: {
    login(auth) {
      this.$store.dispatch('auth/login', auth)
    }
  }
}
</script>
