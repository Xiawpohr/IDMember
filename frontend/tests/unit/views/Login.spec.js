import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Login from '@/views/Login.vue'
import LoginForm from '@/components/LoginForm.vue'

Vue.use(Vuex)
Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper, state, actions, store

beforeEach(() => {
  state = {
    isLoading: false,
    id: null
  }
  actions = {
    login: jest.fn()
  }
  store = new Vuex.Store({
    modules: {
      auth: {
        namespaced: true,
        state,
        actions
      }
    }
  })
  wrapper = shallowMount(Login, {
    localVue: Vue,
    store
  })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should render LoginForm component', () => {
  expect(wrapper.contains(LoginForm)).toBe(true)
})

it('should authenticate user when recieving submitted event', () => {
  const auth = {
    email: 'test@example.com',
    password: '123456'
  }
  const loginForm = wrapper.find(LoginForm)
  loginForm.vm.$emit('submitted', auth)
  expect(actions.login).toHaveBeenCalled()
})
