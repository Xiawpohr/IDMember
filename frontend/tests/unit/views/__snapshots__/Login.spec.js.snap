// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<vlayout-stub tag="div" justify-center="" align-center="">
  <div>
    <loginform-stub></loginform-stub>
    <div class="mt-3 text-xs-center">
      <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" to="/signup" color="teal">
        Go to Sign Up
      </vbtn-stub>
    </div>
  </div>
</vlayout-stub>
`;
